unitTests/test_slice.cpp

```diff
@@ -12,43 +12,23 @@ class slice;
 
 /*!
  * This namespace contains the helper-function get_test_data. It is intented
- * to be used for test with the slice fixture: it returns the appropriate
+ * to be used for tests with the slice fixture: it returns the appropriate
  * data to the constructor of slice. For (const) T==std::vector it returns the
- * fixtures meber vec_, for (const) T==int* it returns vec_.data()
+ * fixtures member vec_, for (const) T==int* it returns vec_.data()
  *
- * Due to C++98's limitations, this requires a separate traits class, that
- * specifies the return type *and* a specialization of get_test_data for each
- * case (maybe some can be reduced with SFINAE, but that ain't improving
- * readability either).
- *
- * Unfortunately, C++11 will probably only make the return_type_traits go away,
- * but not the template specializations of get_test_data (for that we need
- * C++17, so see you in 2025).
+ * As we don't have C++17 and can't use a single function with a big `constexpr
+ * if`, we instead overload get_test_data for the four types that we test.
  */
-namespace cpp_98_boilerplate
+namespace cpp_pre_17_boilerplate
 {
-    template <typename T>
-    struct return_type_traits
-    {
-        typedef T type;
-    };
+    int* get_test_data(slice<int*>& st);
```

Comment:
In most of the code we use _camelCase_ style for naming functions. Was this style choose intended or could we use _camelCase_ here too?

Reply:
I think this is a relic from the past, can't remember the reason.